### DISCUSSION TITLE
Bundle Outdated that's aware of ruby version

### DISCUSSION THREAD

**Discussion by broksonic21** [ruby/rubygems#6767] [General]
Is there a version or field for bundle outdated that can show you all outdated gems, but only the ones that work on your current ruby version? We are finding that bundle outdated shows some gems that only work on newer versions of ruby.

While I agree they are "outdated", it forces these two upgrade paths together; we have some requirements to run on older rubies for customer reasons (I know they are EOL rubies), but I'd like to be as far as updated on gems that I can given that constraint

Example is `brakeman`:

Bundle outdated shows this if running on Ruby 2.7.*:

```
brakeman                     5.4.1    6.0.0   >= 0        development
```

5.4.1 is the last version that works on 2.7*, as 6 requires ruby 3.

**Comment by hsbt** [comment]
`bundle outdated --strict` already does this. With `gem "rbs"` and no version constraint on Ruby 3.2, plain `bundle outdated` offers rbs 4.2.0 and `--strict` reports the bundle up to date, because 4.2.0 requires Ruby 3.3. It filters on your Gemfile requirements as well as the running Ruby, so a pinned version hides newer releases too.
